### ISSUE TITLE
ProfileContext does not seem to work on NMT?

### ISSUE THREAD

**Issue by David-Levinthal** [tensorflow/nmt#275]
TF version R1.5, Cuda 9.1, Cudnn 7.0.4 NMT download from March 9
note: exact same problem occurred with earlier download of NMT
this has been duplicated on V100s and P100s

add the call to ProfileContext to train.py just before the training loop
and indent all the following lines by 2 spaces
this should profile the first 100 steps
  with tf.contrib.tfprof.ProfileContext('/home/levinth/train_dir'
#                                         , trace_steps=range(600, 699),
#                                         dump_steps=[699]
                                                ) as pctx:

    while global_step < num_train_steps:
      ### Run a step ###
      start_time = time.time()
      try:
        step_result = loaded_train_model.train(train_sess)
        hparams.epoch_step += 1
down to:
  # Done training
  loaded_train_model.saver.save(
      train_sess,
this generates a lot of errors when invoked as:
 python -m nmt.nmt --src=de --tgt=en --hparams_path=nmt/standard_hparams/wmt16_gnmt_4_layer_small.json --out_dir=/home/levinth/nmt_temp/deen_gnmt --vocab_prefix=/home/levinth/WMT/vocab.bpe.32000 --train_prefix=/home/levinth/WMT/train.tok.clean.bpe.32000 --dev_prefix=/home/levinth/WMT/newstest2013.tok.bpe.32000 --test_prefix=/home/levinth/WMT/newstest2015.tok.bpe.32000 > deen_gnmt_log_256_cud2_mar9_prof.txt 2>&1
the file wmt16_gnmt_4_layer_small.json is the same as the default but with the hidden size set to 256

the log contains:
  created infer model with fresh parameters, time 0.12s
# Start step 0, lr 1, Mon Mar 19 12:53:10 2018
# Init train iterator, skipping 0 elements
2018-03-19 12:53:18.052126: I tensorflow/stream_executor/dso_loader.cc:147] successfully opened CUDA library libcupti.so.9.1 locally
Node dynamic_seq2seq/decoder/decoder/while/BasicDecoderStep/multi_rnn_cell/cell_0_attention/cell_0_attention/attention/dropout/random_uniform incompatible shapes: Shapes (?, 512) and () are not compatible.
each of these of errors occurs 39 times (in various orders)
Node dynamic_seq2seq/decoder/decoder/while/BasicDecoderStep/multi_rnn_cell/cell_0_attention/cell_0_attention/attention/dropout/random_uniform incompatible shapes: Shapes (?, 512) and () are not compatible.
Node dynamic_seq2seq/encoder/bidirectional_rnn/fw/fw/while/dropout/random_uniform incompatible shapes: Shapes (?, 256) and () are not compatible.
Node dynamic_seq2seq/encoder/bidirectional_rnn/bw/bw/while/dropout/random_uniform incompatible shapes: Shapes (?, 256) and () are not compatible.
Node dynamic_seq2seq/encoder/rnn/while/dropout/random_uniform incompatible shapes: Shapes (?, 512) and () are not compatible.

a profile gets created but appears to only have data from a single step
tfprof> op -select micros,occurrence,float_ops -order_by micros -step 1000
Options -step=1000 not found.
Available steps: 11 

any thoughts on how to get this to work?

**Comment by David-Levinthal** [comment]
the lines in large font like
, trace_steps=range(600, 699),
dump_steps=[699]
are commented out with pound signs..hence the font change
I should have deleted them 
note trying to switch the range from the first 100 steps  to steps 600-699 made things MUCH worse
